WordPress/Classes/Services/BlogService.m

```diff
@@ -185,6 +185,14 @@ - (void)syncBlogAndAllMetadata:(Blog *)blog completionHandler:(void (^)(void))co
         DDLogError(@"Failed to sync Editor settings");
         dispatch_group_leave(syncGroup);
     }];
+    
+    if ([BlazeHelper isBlazeFlagEnabled]) {
+        BlazeService *blazeService = [BlazeService createService];
+        dispatch_group_enter(syncGroup);
+        [blazeService getStatusFor:blog completion:^{
+            dispatch_group_leave(syncGroup);
+        }];
+    }
```

Comment:
[Question] Looking at `AbstractPost.canBlaze`, it seems that `canBlaze` doesn't consider `isBlazeFlagEnabled`, and it depends only on `isBlazeApproved`. Yes, `isBlazeFlagEnabled` controls whether we fetch the blaze status or not. But isn't it possible for the flag to get turned off, and we get stuck in a state where Blaze is enabled in the posts list when it shouldn't? (This doesn't apply to the dashboard and menu entry points because `BlazeHelper.shouldShowCard()` takes `isBlazeFlagEnabled` into consideration)

Reply:
If the above scenario makes sense, then maybe we can change the definition of `AbstractPost.canBlaze` to include `isBlazeFlagEnabled`, but I think a more intuitive solution is the following:
```suggestion
    if ([BlazeHelper isBlazeFlagEnabled]) {
        BlazeService *blazeService = [BlazeService createService];
        dispatch_group_enter(syncGroup);
        [blazeService getStatusFor:blog completion:^{
            dispatch_group_leave(syncGroup);
        }];
    } else {
        blog.isBlazeApproved = false;
    }
```

Reply:
Or better yet, we can move the `[BlazeHelper isBlazeFlagEnabled]` check inside `BlazeService.getStatus(for:completion:)`